hypothesis_2.ipynb

```diff
@@ -199,22 +199,33 @@
     "    \n",
     "for row in users_lat:\n",
     "    username = row['username']\n",
+    "    # AW: It's extremely unsafe to evaluate external inputs.  Instead,\n",
```

Comment:
@adamwight 
It is extremely unsafe to evaluate external input overall, but his expression will only recognize a certain set of Python literals and the only disadvantage is that it can ”crash the Python interpreter due to stack depth limitations in Python’s AST compiler. ir raise [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError), [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError), [MemoryError](https://docs.python.org/3/library/exceptions.html#MemoryError) and [RecursionError](https://docs.python.org/3/library/exceptions.html#RecursionError) depending on the malformed input.” Which, for our case, will not be a problem, since we are pre-processing the file, I have control over the file and is not a “free input” and even if it was malformed input, the app would just throw an exception, as it should. Docs: https://docs.python.org/3/library/ast.html#ast.literal_eval.


Based on this arguments, I am keeping it. Unless there is another consideration you think is important to add for safety.

Reply:
Thanks for the pointer, +1 that it does seem mostly safe.  Then my concern is just about using such a non-standard data format that would be difficult to read with anything other than python.   We should use JSON or raw delimiter-joined instead, IMHO.